### ISSUE TITLE
Impossible to load snapshot

### ISSUE THREAD

**Issue by curquiza** [meilisearch/MeiliSearch#958]
## Steps

In an empty directory only containing the MeiliSearch binary (v0.14.0)

1. I launch MeiliSearch with a snapshot creation every 10 secondes

```bash
$ ./meilisearch --snapshot-path mySnapShots/ --snapshot-interval-sec 10
```

2. I create an index and I add documents. But the bug is also present even if I do not do any action with MeiliSearch.

3. After one snapshot was created, I stop MeiliSearch. Two folders are present: `data.ms` and `mySnapShots/` (containing `data.ms.tar.gz`)

4. I try to launch MeiliSearch with the snapshot:
```bash
$ ./meilisearch --load-from-snapshot mySnapShots/data.ms.tar.gz
Error: database already exists at "./data.ms"
```

=> the error is expected [according to the docs](https://docs.meilisearch.com/guides/advanced_guides/safeguards.html#start-from-snapshot).

5. So I remove `data.ms`

6. I re-launch MeiliSearch with the same command:
```bash
$ ./meilisearch --load-from-snapshot mySnapShots/data.ms.tar.gz
Error: Cannot open database, expected MeiliSearch engine version: <0.12.0, current engine version: 0.14.0
```
=> a `data.ms` is now present in my folder.
=> **the error is not expected**.

7. If I try to use the `--ignore-snapshot-if-db-exists` flag:

```bash
./meilisearch --load-from-snapshot mySnapShots/data.ms.tar.gz --ignore-snapshot-if-db-exists
error: The argument '--ignore-snapshot-if-db-exists <ignore-snapshot-if-db-exists>' requires a value but none was supplied

USAGE:
    meilisearch --db-path <db-path> --env <env> --http-addr <http-addr> --http-payload-size-limit <http-payload-size-limit> --ignore-snapshot-if-db-exists <ignore-snapshot-if-db-exists> --load-from-snapshot <load-from-snapshot> --max-mdb-size <max-mdb-size> --max-udb-size <max-udb-size> --no-analytics <no-analytics> --sentry-dsn <sentry-dsn>

For more information try --help
```

=> **This flag should not expect any value**.

## Version

MeiliSearch v0.14.0

## Fix

It should be a hotfix with a `v0.14.1` since the snapshots are not usable right now (If I used them correctly)

**Comment by bidoubiwa** [comment]
I confirm this bug as I did not work for me aswell.
```bash
➜  meili-binaries ./meilisearch --snapshot-path plouf --snapshot-interval-sec 10
# MeiliSearch Launches
➜  meili-binaries ls
data.ms       meilisearch   plouf
➜  meili-binaries ./meilisearch --load-from-snapshot plouf/data.ms.tar.gz
Error: database already exists at "./data.ms"
➜  meili-binaries rm -rf data.ms
➜  meili-binaries ./meilisearch --load-from-snapshot plouf/data.ms.tar.gz
Error: Cannot open database, expected MeiliSearch engine version: <0.12.0, current engine version: 0.14.0
➜  meili-binaries ./meilisearch --snapshot-path plouf/ --load-from-snapshot plouf/data.ms.tar.gz
Error: database already exists at "./data.ms"
➜  meili-binaries rm -rf data.ms
➜  meili-binaries ./meilisearch --snapshot-path plouf/ --load-from-snapshot plouf/data.ms.tar.gz
Error: Cannot open database, expected MeiliSearch engine version: <0.12.0, current engine version: 0.14.0
```

**Comment by MarinPostma** [comment]
There is nothing much we can do right now. This is the expected behavior from `clap` apparently https://github.com/TeXitoi/structopt/issues/428#event-3739844591

There is a conflict with the fact that the flag is also fetched from the environment, and expecting a value (`true`/ `false`).

We have to rethink our option system, I suggest this be part of #907, as we'll have to re-define setting precedence anyway. In the meantime, you'll have to pass `true` to these options to enable them.

Note that this behavior is the same for all options flags, and must have been for some time now (the issue is > 1 year old).

On quick fix would be to remove the environment variables for these flags, but that would be a regression.

**Comment by curquiza** [comment]
Ok, good to know, we need to change the docs then 🙂 @bidoubiwa

**Comment by Kerollmops** [comment]
There is an issue [that has been added to the 3.0 milestone of clap on 11 of july](https://github.com/clap-rs/clap/issues/1476#event-3536418605), maybe that will be fixed one day.

**Comment by MarinPostma** [comment]
closed in favor of #907
